s3/s3.go

```diff
@@ -0,0 +1,233 @@
+	}
+	level, ok := argv["<level>"].(string)
+	if !ok {
+		fmt.Println("Invalid level format.")
+		return false
+	}
+	region, ok := argv["<region>"].(string)
+	if !ok {
+		fmt.Println("Invalid region format.")
+		return false
+	}
+
+	// Load configuration
+	config, err := config.Load()
+	if err != nil {
+		fmt.Println("Failed to load configuration file, error: ", err)
+		return false
+	}
+
+	// Set credentials for auth
+	myAuth := auth.New(
+		&tcclient.Credentials{
+			ClientID:    config["config"]["clientId"].(string),
+			AccessToken: config["config"]["accessToken"].(string),
+			Certificate: config["config"]["certificate"].(string),
+		},
+	)
+
+	// Get credential for AWS S3
+	resp, err := myAuth.AwsS3Credentials(level, bucket, prefix)
+	if err != nil {
+		fmt.Println("Failed to load AWS S3 credentials: ", err)
+		return false
+	}
+
+	// Set Credentials for AWS S3
+	aws_access_key_id := resp.Credentials.AccessKeyID
+	aws_secret_access_key := resp.Credentials.SecretAccessKey
+	aws_session_token := resp.Credentials.SessionToken
+	//aws_expires := &resp.Expires
+
+	creds := credentials.NewStaticCredentials(aws_access_key_id, aws_secret_access_key, aws_session_token)
+	_, err = creds.Get()
+	if err != nil {
+		fmt.Printf("Invalid credentials: %s", err)
+		return false
+	}
+
+	// Create a session which contains the configurations for the SDK.
+	// Use the session to create the service clients to make API calls to AWS.
+	aws_config := aws.NewConfig().WithRegion(region).WithCredentials(creds)
+	sess := session.New(aws_config)
+
+	// For upload
+	if argv["upload"] == true {
+
+		// Parse for filename
+		filename, ok := argv["<file>"].(string)
+		if !ok {
+			fmt.Println("Invalid file format.")
+			return false
+		}
+
+		// Create service client
+		svc := awsS3Manager.NewUploader(sess)
+
+		// Open the filename and return the file
+		file, err := os.Open(filename)
+		if err != nil {
+			fmt.Println("Failed to open file", filename, err)
+			return false
+		}
+		defer file.Close()
+
+		// Generate key from filename
+		key := filepath.Base(filename)
+
+		fmt.Printf("Uploading %s to s3://%s/%s...\n", key, bucket, prefix)
+
+		// Upload the file
+		params := &awsS3Manager.UploadInput{
+			Bucket: aws.String(bucket),
+			Key:    aws.String(key),
+			Body:   file,
+			//Expires: aws_expires,
+		}
+		_, err = svc.Upload(params)
+		if err != nil {
+			if multierr, ok := err.(awsS3Manager.MultiUploadFailure); ok {
+				// Process error and its associated uploadID for Multipart Upload failure
+				fmt.Println("Multipart Upload error: ", multierr.Code(), multierr.Message(), multierr.UploadID())
+				return false
+			} else {
+				// Process error generically
+				fmt.Printf("Failed to upload %s to s3://%s/%s, %s\n", key, bucket, prefix, err.Error())
+				return false
+			}
+		}
+
+		fmt.Printf("Successfully uploaded %s to s3://%s/%s\n", key, bucket, prefix)
+
+	} else if argv["download"] == true {
+
+		// Parse for key
+		key, ok := argv["<key>"].(string)
```

Comment:
I suggest that the upload and download is split into separate functions... This just changes the code organization, but it could simplify the code a lot, reduce levels of indentation (improving code quality), and if we later decide to add sub-command like list, copy, delete then the code won't be horribly complicated.

 Maybe `Execute` should just parse arguments... Get credentials, detect bucket region and setup s3 client library... Then the file operations might be better to do in a separate method, maybe in a separate file.